File: MiniDOMFinanziaria/Finanziaria/Pratiche/Statistiche/ClientiLavorati/ClientiLavoratiFilter.vb
Imports minidom
Imports minidom.Databases
Imports minidom.Sistema
Imports minidom.Anagrafica
Imports minidom.CustomerCalls

Partial Public Class Finanziaria

     
    <Serializable> _
    Public Class ClientiLavoratiFilter
        Implements XML.IDMDXMLSerializable

        Public IDPuntoOperativo As Integer
        Public IDConsulente As Integer
        Public DataInizio As Date?
        Public DataFine As Date?
        Public TipoFonte As String
        Public IDFonte As Integer

        Public Sub New()
            DMDObject.IncreaseCounter(Me)
            Me.IDPuntoOperativo = 0
            Me.IDConsulente = 0
            Me.DataInizio = Nothing
            Me.DataFine = Nothing
            Me.TipoFonte = ""
            Me.IDFonte = 0
        End Sub



        Public Sub SetFieldInternal(fieldName As String, fieldValue As Object) Implements XML.IDMDXMLSerializable.SetFieldInternal
            Select Case fieldName
                Case "IDPuntoOperativo" : Me.IDPuntoOperativo = XML.Utils.Serializer.DeserializeInteger(fieldValue)
                Case "IDConsulente" : Me.IDConsulente = XML.Utils.Serializer.DeserializeInteger(fieldValue)
                Case "DataInizio" : Me.DataInizio = XML.Utils.Serializer.DeserializeDate(fieldValue)
                Case "DataFine" : Me.DataFine = XML.Utils.Serializer.DeserializeDate(fieldValue)
                Case "TipoFonte" : Me.TipoFonte = XML.Utils.Serializer.DeserializeString(fieldValue)
                Case "IDFonte" : Me.IDFonte = XML.Utils.Serializer.DeserializeInteger(fieldValue)
            End Select
        End Sub

        Public Sub XMLSerialize(writer As XML.XMLWriter) Implements XML.IDMDXMLSerializable.XMLSerialize
            writer.WriteAttribute("IDPuntoOperativo", Me.IDPuntoOperativo)
            writer.WriteAttribute("IDConsulente", Me.IDConsulente)
            writer.WriteAttribute("DataInizio", Me.DataInizio)
            writer.WriteAttribute("DataFine", Me.DataFine)
            writer.WriteAttribute("TipoFonte", Me.TipoFonte)
            writer.WriteAttribute("IDFonte", Me.IDFonte)
        End Sub

        Protected Overrides Sub Finalize()
            MyBase.Finalize()
            DMDObject.DecreaseCounter(Me)
        End Sub
    End Class




End Class

File: MiniDOMFinanziaria/Finanziaria/Pratiche/Statistiche/ClientiLavorati/ClientiLavoratiFilter.cs
using System;
using DMD;
using DMD.XML;
using minidom;
using static minidom.Sistema;

namespace minidom
{
    public partial class Finanziaria
    {
        [Serializable]
        public class ClientiLavoratiFilter
            : IDMDXMLSerializable
        {
            public int IDPuntoOperativo;
            public int IDConsulente;
            public DateTime? DataInizio;
            public DateTime? DataFine;
            public string TipoFonte;
            public int IDFonte;

            public ClientiLavoratiFilter()
            {
                DMDObject.IncreaseCounter(this);
                IDPuntoOperativo = 0;
                IDConsulente = 0;
                DataInizio = default;
                DataFine = default;
                TipoFonte = "";
                IDFonte = 0;
            }

            public void SetFieldInternal(string fieldName, object fieldValue)
            {
                switch (fieldName ?? "")
                {
                    case "IDPuntoOperativo":
                        {
                            IDPuntoOperativo = (int)DMD.XML.Utils.Serializer.DeserializeInteger(fieldValue);
                            break;
                        }

                    case "IDConsulente":
                        {
                            IDConsulente = (int)DMD.XML.Utils.Serializer.DeserializeInteger(fieldValue);
                            break;
                        }

                    case "DataInizio":
                        {
                            DataInizio = DMD.XML.Utils.Serializer.DeserializeDate(fieldValue);
                            break;
                        }

                    case "DataFine":
                        {
                            DataFine = DMD.XML.Utils.Serializer.DeserializeDate(fieldValue);
                            break;
                        }

                    case "TipoFonte":
                        {
                            TipoFonte = DMD.XML.Utils.Serializer.DeserializeString(fieldValue);
                            break;
                        }

                    case "IDFonte":
                        {
                            IDFonte = (int)DMD.XML.Utils.Serializer.DeserializeInteger(fieldValue);
                            break;
                        }
                }
            }

            public void XMLSerialize(XMLWriter writer)
            {
                writer.WriteAttribute("IDPuntoOperativo", IDPuntoOperativo);
                writer.WriteAttribute("IDConsulente", IDConsulente);
                writer.WriteAttribute("DataInizio", DataInizio);
                writer.WriteAttribute("DataFine", DataFine);
                writer.WriteAttribute("TipoFonte", TipoFonte);
                writer.WriteAttribute("IDFonte", IDFonte);
            }

            ~ClientiLavoratiFilter()
            {
                DMDObject.DecreaseCounter(this);
            }
        }
    }
}